src/ApiProblem/ApiProblem.php

```diff
@@ -135,7 +135,9 @@ public static function fromException(Exception $exception, $title = '', $type =
             $type = self::RFC2616;
         }
 
-        return new self($code, $exception->getMessage(), $title, $type, $additionalDetails);
+        $detail = empty($exception->getMessage()) ? get_class($exception) : $exception->getMessage();
```

Comment:
I like the idea but I've got some concerns. 

First I would not leak the fully qualified namespace. How about only the exception name? 

Yet, this brings us to a second case. Still I find it risky because for instance capturing a PDOException and returning the class name reveals it's SQL problem (query, timeout, locked table, etc).